appveyor.yml

```diff
@@ -58,6 +68,53 @@ on_failure:
 artifacts:
   - path: dist\*
 
+# scripts to run before deployment
+before_deploy:
+  # install gravitybee/generate the standalone executable
+  - mkdir C:\venv
+  - C:\Python36-x64\Scripts\virtualenv C:\venv
+  - C:\venv\Scripts\activate
+  - python --version
+  - pip install gravitybee satsuki
+  - pip install --editable .
+  - gravitybee --src-dir src --sha file --extra-data static --verbose
+  # prepare deployment
+  - set SATS_TAG=$gb_pkg_ver
+  - set SATS_BODY=* [$gb_pkg_name v$gb_pkg_ver CHANGELOG](https://github.com/plus3it/$gb_pkg_name_lower/blob/$gb_pkg_ver/CHANGELOG.md) (Win)
+  - set SATS_REL_NAME=Release v$gb_pkg_ver
+  - set LOCAL_STAGING_DIR=s3_uploads
+  - ps: >-
+      if($env:APPVEYOR_REPO_BRANCH -eq "master") {
+        $env:S3_BUCKET=$env:RELEASE_BUCKET
+      } else {
+        $env:S3_BUCKET=$env:DEV_BUCKET
+      }
+  - set S3_KEYFIX=releases
+  - gravitybee-environs.bat
+  # upload & rename artifacts to appveyor's magic artifact depot in the sky (DITS)
+  - appveyor PushArtifact %GB_ENV_GEN_FILE_W_PATH% -FileName %GB_ENV_APP_VERSION%\%GB_ENV_GEN_FILE%
+  - appveyor PushArtifact %GB_ENV_APP_NAME%-%GB_ENV_APP_VERSION%-sha256-%GB_ENV_OPERATING_SYSTEM%-%GB_ENV_MACHINE_TYPE%.json -FileName %GB_ENV_APP_VERSION%\%GB_ENV_APP_NAME%-%GB_ENV_APP_VERSION%-sha256-%GB_ENV_OPERATING_SYSTEM%-%GB_ENV_MACHINE_TYPE%.json
+  - appveyor PushArtifact %GB_ENV_GEN_FILE_W_PATH% -FileName latest\%GB_ENV_APP_NAME%-latest-standalone-%GB_ENV_OPERATING_SYSTEM%-%GB_ENV_MACHINE_TYPE%.exe
+  - appveyor PushArtifact %GB_ENV_APP_NAME%-%GB_ENV_APP_VERSION%-sha256-%GB_ENV_OPERATING_SYSTEM%-%GB_ENV_MACHINE_TYPE%.json -FileName latest\%GB_ENV_APP_NAME%-latest-sha256-%GB_ENV_OPERATING_SYSTEM%-%GB_ENV_MACHINE_TYPE%.json
+  - SET
+  - ps: ls
+  - ps: >-
+      if($env:APPVEYOR_REPO_BRANCH -eq "master") {
+        satsuki --verbose
+      }
+deploy:
+  - provider: S3
+    access_key_id:
+      secure: IgUARv6O1DqRDRKQ3rjZEuDdPGuzz/QCzjkwdWRqx5o=
+    secret_access_key:
+      secure: RiIvGJ442UiwnoeZ53D5mpLKXwn/e5PSEYxZR/FIL4cFIRkbnZTzGMiuMoEbXRcE
+    bucket: $(S3_BUCKET)
+    folder: releases
+    #artifact: # not needed since by default it will grab everything added to DITS
+    on:
+      do_deploy: yes
```

Comment:
I think the workaround adds more code than a second deploy provider with a different condition, especially if the access/secret keys are set as envs in the service rather than the file... 😆 

It's declarative vs imperative styles. For these configs, declarative is much easier to read.

Reply:
I'm non-denominational when it comes to declarative vs. imperative.